lib/query/querycompiler.js

```diff
@@ -1199,7 +1199,13 @@ class QueryCompiler {
       let tableName = this.single.table;
       const schemaName = this.single.schema;
 
-      if (tableName && schemaName) tableName = `${schemaName}.${tableName}`;
+      const isQueryBuilder = tableName instanceof QueryBuilder;
+      const isRawQuery = tableName instanceof Raw;
+      const isFunction = typeof tableName === 'function';
+
+      if (tableName && schemaName && !isQueryBuilder && !isRawQuery && !isFunction) {
```

Comment:
Since last three checks are only relevant if first three pass, I would split checks in two blocks and only resolve these conditions if first two pass.

Reply:
Refactored